### DISCUSSION TITLE
pnpm support

### DISCUSSION THREAD

**Discussion by aparajita** [ionic-team/capacitor#3355] [Ideas]
Currently Capacitor assumes there are only two package managers in the world: npm and yarn. It would be nice if it would support [pnpm](https://pnpm.js.org) as well. `pnpm` support is in Ionic already.

**Comment by imhoffd** [comment]
We're using https://github.com/ionic-team/capacitor/issues/1863 to track this feature request. Thanks!

**Reply by aparajita** [reply]
I'm surprised stuff isn't breaking with yarn 2, because you are relying on transitive dependencies, which I believe are disallowed in yarn 2. Eventually I'm sure npm will follow suit.

It isn't the greatest idea to rely on the internal structure of node_modules, which **currently** is a flattened hierarchy with npm and yarn 1. That easily could change (and it has in the past), as it is completely undocumented and unspecified.

**Comment by lexarion1** [comment]
Hello. Any plans to return to this topic? pnpm is very popular and fast.

**Reply by jcesarmobile** [reply]
Capacitor doesn't require any package managers as it doesn't install packages anymore, you should be able to use any package manager.
